zfs.go

```diff
@@ -244,9 +251,8 @@ func (z *snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 	err = t.Commit()
 	t = nil
 	if err != nil {
-		snapshotName := activeName + "@" + snapshotSuffix
 		if derr := destroySnapshot(snapshot); derr != nil {
-			log.G(ctx).WithError(derr).WithField("snapshotName", snapshotName).Error("failed to delete dataset")
+			log.G(ctx).WithError(derr).WithField("snapshotName", snapshot.Name).Error("failed to delete dataset")
```

Comment:
Is it worth calling this out as `failed to delete snapshot`?

Reply:
removed this part. can be another PR.